src/main/java/teammates/common/util/StringHelper.java

```diff
@@ -359,6 +359,177 @@ public static String csvToHtmlTable(String str) {
                              result.toString());
     }
 
+    /**
+     * Combines all the strings inside the passed list using the specified delimiter. The result will be
+     * another list with a single entry, which is the combined string, or an empty list if the passed
+     * argument is an empty list.
+     * @param list of string
+     * @param delimiter the connectors in between the strings
+     * @return list containing the combined string or an empty list
+     */
+    private static List<String> combineListEntries(List<String> list,
+            String delimiter) {
+        if (list.isEmpty()) {
+            return list;
+        } else {
+            String combinedEntries = list.get(0);
+            List<String> result = new ArrayList<String>();
+            if (list.size() > 1) {
+                for (int i = 1; i < list.size(); i++) {
+                    combinedEntries += delimiter + list.get(i);
+                }
+            }
+            result.add(combinedEntries);
+            return result;
+        }
+    }
+    
+    /**
+     * If the content of the rowData is one of the following, they will be processed
+     * accordingly (for the sake of aesthetics):<br><ol>
+     * <li> Course, coursename -> Course: coursename</li>
+     * <li> Session Name, sessionname -> Session Name: sessionname</li>
+     * <li> Question #, qntitle -> Question #: qntitle</li>
+     * <li> Summary Statistics, (empty) -> Summary statistics</li>
+     * <li> In the points given below, (...) (comma here is the column separator) ->
+     * In the points given below, (...) (comma here is the punctuation comma)</li>
+     * </ol>
+     * 4 and 5 need to be done as they are unintended consequences of comma-separating algorithm.
+     * 
+     * @param rowData
+     * @return processed list of string
+     */
+    private static List<String> processSpecialKeyword(List<String> rowData) {
```

Comment:
It is not clear what this function does as a glance. "processSpecialKeywords" does not make much sense in this context (StringHelper.java).
The conditionals below could be changed to make clear what each branch is checking for, by adding comments or introducing boolean variables.
Another concern is what happens if the format of the csv changes? and rowData.get(0) is changed?